> LGTM. @rjrock But can you attach the link the AMD CI that this is going to fix and what status we should expect? I will add the ready label.

Yes, this PR partially addresses https://github.com/vllm-project/vllm/issues/37709. In particular the test
```bash
python3 examples/offline_inference/data_parallel.py --model=Qwen/Qwen1.5-MoE-A2.7B -tp=1 -dp=2 --max-model-len=2048 --all2all-backend=deepep_high_throughput
```

You can see in the nightly CI the failure, https://buildkite.com/vllm/amd-ci/builds/7612/steps/canvas?sid=019d8a93-8a56-40f5-9d3b-d4a45284fa79&tab=output#L2327-L2338, that this test fails
```text
(EngineCore_DP0 pid=2615) RuntimeError: Worker failed with error 'Failed: CUDA error /app/DeepEP/csrc/kernels/launch_hip.cuh:71 'invalid kernel file'', please check the stack trace above for the root cause
--
(EngineCore_DP0 pid=2615) DEBUG 04-14 06:32:27 [distributed/device_communicators/shm_broadcast.py:168] Canceling waiting reads on SHM Buffer
DEBUG 04-14 06:32:28 [v1/engine/utils.py:1143] Waiting for 1 local, 0 remote core engine proc(s) to start.
DEBUG 04-14 06:32:28 [v1/engine/utils.py:1143] Waiting for 1 local, 0 remote core engine proc(s) to start.
Process Process-1:
Traceback (most recent call last):
  File "/usr/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/usr/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/vllm-workspace/examples/offline_inference/data_parallel.py", line 137, in main
    llm = LLM(**engine_args)
```

Updating the DeepEP version to one with a setup.py that correctly builds the fat binary makes the test succeed and yields the output
> DP rank 0, Prompt: 'Hello, my name is', Generated text: ' Josh and I will be the teacher for the Braille and Computers class for this'
> DP rank 0, Prompt: 'The president of the United States is', Generated text: ' the most powerful person in the world. The President is the head of the executive'
> DP rank 0, Prompt: 'The capital of France is', Generated text: '______.\nA. London\nB. Paris\nC. New York\n'
> DP rank 0, Prompt: 'The future of AI is', Generated text: ' being decided in Cambridge\nArtificial intelligence (AI) is one of the most'
> DP rank 0, Prompt: 'Hello, my name is', Generated text: ' Belinda and I am a 43 year old woman who is passionate about'
> DP rank 1, Prompt: 'Hello, my name is', Generated text: ' Josh and I will be the teacher for the Braille and Literacy Teaching in the 21'
> DP rank 1, Prompt: 'The president of the United States is', Generated text: ' the most powerful person in the world. The President is the head of the executive branch of the U'
> DP rank 1, Prompt: 'The capital of France is', Generated text: ' ______.\nA. Berlin\nB. London\nC. Madrid\nD. Paris\n答案:\n'
> DP rank 1, Prompt: 'The future of AI is', Generated text: ' being determined right now – by you.\nFor all the excitement about the transformative power of artificial intelligence,'
> DP rank 1, Prompt: 'Hello, my name is', Generated text: ' Alan Belcher and I am a 43 year old male. I am a 20'

--- a/.buildkite/test-amd.yaml
+++ b/.buildkite/test-amd.yaml
@@ -2613,6 +2613,7 @@ steps:
   - vllm/platforms/rocm.py
   commands:
   - export TORCH_NCCL_BLOCKING_WAIT=1
+  - VLLM_LOGGING_LEVEL=DEBUG python3 examples/offline_inference/data_parallel.py --model=Qwen/Qwen1.5-MoE-A2.7B -tp=1 -dp=2 --max-model-len=2048 --all2all-backend=deepep_high_throughput
   - pytest -v -s tests/v1/distributed/test_dbo.py
 
 
@@ -3601,7 +3602,6 @@ steps:
   commands:
   - export TORCH_NCCL_BLOCKING_WAIT=1
   - pytest -v -s tests/distributed/test_context_parallel.py
-  - VLLM_LOGGING_LEVEL=DEBUG python3 examples/offline_inference/data_parallel.py --model=Qwen/Qwen1.5-MoE-A2.7B -tp=1 -dp=2 --max-model-len=2048 --all2all-backend=deepep_high_throughput
   - pytest -v -s tests/v1/distributed/test_dbo.py
 
 

--- a/docker/Dockerfile.rocm
+++ b/docker/Dockerfile.rocm
@@ -192,9 +192,10 @@ RUN cd /opt/rixl && mkdir -p /app/install && \
 FROM base AS build_deep
 ARG ROCSHMEM_BRANCH="ba0bf0f3"
 ARG ROCSHMEM_REPO="https://github.com/ROCm/rocm-systems.git"
-ARG DEEPEP_BRANCH="e84464ec"
+ARG DEEPEP_BRANCH="5d90af8b"
 ARG DEEPEP_REPO="https://github.com/ROCm/DeepEP.git"
 ARG DEEPEP_NIC="cx7"
+ARG DEEPEP_ROCM_ARCH="gfx942;gfx950"
 ENV ROCSHMEM_DIR=/opt/rocshmem
 
 RUN git clone ${ROCSHMEM_REPO} \
@@ -202,13 +203,12 @@ RUN git clone ${ROCSHMEM_REPO} \
  && git checkout ${ROCSHMEM_BRANCH} \
  && mkdir -p projects/rocshmem/build \
  && cd projects/rocshmem/build \
- && cmake .. \
-    -DCMAKE_INSTALL_PREFIX="${ROCSHMEM_DIR}" \
-    -DROCM_PATH=/opt/rocm \
-    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
-    -DUSE_EXTERNAL_MPI=OFF \
- && make -j \
- && make install
+ && bash ../scripts/build_configs/all_backends \
+      -DCMAKE_INSTALL_PREFIX="${ROCSHMEM_DIR}" \
+      -DROCM_PATH=/opt/rocm \
+      -DGPU_TARGETS="gfx942;gfx950" \
+      -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+      -DUSE_EXTERNAL_MPI=OFF
 
 # Build DeepEP wheel.
 # DeepEP looks for rocshmem at ROCSHMEM_DIR.

--- a/docker/Dockerfile.rocm
+++ b/docker/Dockerfile.rocm
@@ -206,7 +206,7 @@ RUN git clone ${ROCSHMEM_REPO} \
  && bash ../scripts/build_configs/all_backends \
       -DCMAKE_INSTALL_PREFIX="${ROCSHMEM_DIR}" \
       -DROCM_PATH=/opt/rocm \
-      -DGPU_TARGETS="gfx942;gfx950" \
+      -DGPU_TARGETS="${DEEPEP_ROCM_ARCH}" \
       -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
       -DUSE_EXTERNAL_MPI=OFF
 

--- a/docker/Dockerfile.rocm
+++ b/docker/Dockerfile.rocm
@@ -207,7 +207,6 @@ RUN git clone ${ROCSHMEM_REPO} \
       -DCMAKE_INSTALL_PREFIX="${ROCSHMEM_DIR}" \
       -DROCM_PATH=/opt/rocm \
       -DGPU_TARGETS="${DEEPEP_ROCM_ARCH}" \
-      -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
       -DUSE_EXTERNAL_MPI=OFF
 
 # Build DeepEP wheel.

